Improve/md quickstart

edison23 opened the pull request:
Updated the midPoint/Quickstart guide to make it easier to follow. I also lowered TOC max-depth to make it list only the important hilvl sections to make the TOC easier to scan. Please review my changes.

licehammer commented on midpoint/quickstart/index.adoc:
```diff
@@ -1,136 +1,116 @@
+
+The script is built for *Linux*, and it also works on macOS.
+If you're running another operating system, you may spend more time adjusting your system to satisfy the requirements.
+Another option is to install Linux on a virtual machine (e.g., using link:https://www.virtualbox.org/[VirtualBox]) and run the script on the virtual machine.
```
For other OS, just send users to our docker docs.

licehammer commented on midpoint/quickstart/index.adoc:
```diff
@@ -1,136 +1,116 @@
+Usually, if you're on Linux or macOS, no extra step is needed here.
+This script is written so that it works on all modern Linux and macOS systems.
+
+If you're on Windows, you'll need to use the link:https://learn.microsoft.com/en-us/windows/wsl/about[Windows Subsystem for Linux (WSL)] or run Linux in a virtual machine.
```
Windows is already covered in the beginning of this guide.

licehammer commented on midpoint/quickstart/index.adoc:
```diff
@@ -197,164 +199,170 @@ Clean up process done.
+| Specify the initial password for the administrator user object.
+Keep the <<password-policy,password policy>> in mind. +
+Intended for the first run only.
+Don't use for password change once the user is created.
```
This line won't be most likely clear for new users.

licehammer commented on midpoint/quickstart/index.adoc:
```diff
@@ -197,164 +199,170 @@ Clean up process done.
+
+| `-subdir`
+| A comma-separated list of directories to be created during the initialization. +
+This is useful when you need additional directories, e.g., *exports*.
```
This is also confusing for new users. (I know this pull request is just reformatting this part, but it should be improved in the future.)

edison23 replied:
Could you, please, suggest some simple use case / scenario for which users might need this option? I realized I'm not entirely clear on what exactly this is to be used for and what it entails (e.g., will it also create auto-mounted volumes for the container?). Thanks!

licehammer commented on midpoint/quickstart/index.adoc:
```diff
@@ -13,9 +13,8 @@ To make the deployment as easy as possible for you, we provide a shell script yo
-If you're running another operating system, you may spend more time adjusting your system to satisfy the requirements.
-Another option is to install Linux on a virtual machine (e.g., using link:https://www.virtualbox.org/[VirtualBox]) and run the script on the virtual machine.
+The script is built for *Linux* and it also works on macOS.
+If you're running another operating system, you may spend more time adjusting your system to make the script useful to you.
```
This is intended for beginners. We can't expect them to adjust the script. For Windows we can suggest docker desktop, that is yet to be documented.

licehammer commented on midpoint/quickstart/index.adoc:
```diff
@@ -25,8 +24,8 @@ Before you proceed further, make sure the following requirements are satisfied o
-You need to have Docker installed on your computer to follow through with this guide.
-Check the link:https://docs.docker.com/engine/install/[official Docker documentation] for guides on installing Docker.
+You need to have Docker installed on your computer.
+Check the documentation on link:https://docs.docker.com/engine/install/[Docker Engine] and link:https://docs.docker.com/desktop/setup/install/windows-install/[Docker Desktop] for guides on setting up Docker.
```
Installation of docker desktop for Windows is not relevant since this script is only for Linux and Mac.

licehammer commented on midpoint/quickstart/index.adoc:
```diff
@@ -301,14 +297,15 @@ Use this attribute to change the required version without the need to specify th
 proper password.
 
 If you keep the generated initial password, we recommend changing it after the system is properly set up anyway.
+See xref:/midpoint/reference/admin-gui/self-service[midPoint self-service UI] for details.
```
This link is not relevant.

edison23 commented:
@licehammer : Thanks for your reviews. I addressed your remarks, the only issue that remains is the `-subdit` parameter, use of which I don't fully understand - I left that thread above open but I intend to merge this PR none the less.
